### PR TITLE
BJS2-15673 - Кэширование достижений

### PR DESCRIPTION
**Pull request by Kostahe**
- **Added `AchievementCache` Component**:
  - Added `init()` method to initialize cache with achievements from the repository.
  - Implemented caching and retrieval of achievements with JSON serialization and deserialization.

- **Added `AchievementController`**:
  - Implemented endpoint to retrieve achievements by title.

- **Added `AchievementDto`**:
  - Defined `AchievementDto` record for data transfer with relevant fields.

- **Added `AchievementMapper`**:
  - Created `AchievementMapper` interface for mapping between `Achievement` and `AchievementDto`.

- **Added `AchievementService`**:
  - Implemented service logic to handle retrieval of achievements from cache or repository.

- **Added Global Exception Handling**:
  - Implemented `GlobalExceptionHandler` to handle `JsonParseException` and `EntityNotFoundException`.
  - Added `ErrorResponseDto` for consistent error responses.

- **Added Configurations**:
  - Added `ObjectMapperConfig` for JSON configuration.
  - Configured `RedisConfig` for Redis connection setup.

### CODE REVIEW

File: src/main/java/faang/school/achievement/cache/AchievementCache.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    public Optional<Achievement> getAchievementByTitle(String title) {
+        String json = (String) redisTemplate.opsForHash().get(achievementKey, title);
+        return json != null ? Optional.of(deserialize(json)) : Optional.empty();
+    }
+
+    private void cacheAchievement(Achievement achievement) {
+        String json = serialize(achievement);
+        redisTemplate.opsForHash().put(achievementKey, achievement.getTitle(), json);
+    }
```
Comment (pablo023) [inline src/main/java/faang/school/achievement/cache/AchievementCache.java:42]: Неплохо, но надо бы ттл установить? А то там мусорка в итоге соберется

**Review by pablo023 — APPROVED** [review]
Это сделано неплохо. В свободное время почитай про Redis Sentinel, и про TTL.

**Comment by pablo023** [comment]
Исправь конфликт, и можешь смержить.

File: src/main/java/faang/school/achievement/cache/AchievementCache.java (hunk 1 above)
Comment (Kostahe) [reply src/main/java/faang/school/achievement/cache/AchievementCache.java:42]: Окей посмотрю

**Pull request merged**